src/lib.rs

```diff
@@ -169,15 +169,32 @@ pub struct Interpreter<'a> {
 }
 
 impl<'a> Interpreter<'a> {
-    fn new<R: Read, W: Write>(code: String, input: &'a mut R, output: &'a mut W) -> Interpreter<'a> {
-        Interpreter {
-            code: code,
+    /// Return a new interpreter with the given code, reader, and writter.
+    ///
+    /// Interpreters are relatively large, so avoid too many calls to this
+    /// function.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use brainfuck::Interpreter;
+    ///
+    /// let mut reader = &[][..];
+    /// let mut writer = Vec::<u8>::new();
+    /// Interpreter::new("+.", &mut reader, &mut writer).unwrap().run();
+    /// ```
+    pub fn new<C: Into<String>, R: Read, W: Write>(code: C, input: &'a mut R, output: &'a mut W) -> Result<Interpreter<'a>, Error> {
```

Comment:
I'm still getting the hang on what naming conventions I like in Rust in some cases. In this case the argument is as follows.

1. `new` is often treated as the constructor or canonical way to create an object.
2. Since this actually takes an `Into<String>` the name `from_string` is a bit of a misnomer.